src/Definition/Doc.elm

```diff
@@ -39,16 +41,21 @@ import Html
         )
 import Html.Attributes exposing (alt, class, href, id, rel, src, start, target, title)
 import Html.Events exposing (onClick)
-import Id exposing (Id)
 import Json.Decode as Decode exposing (bool, field, index, int, string)
-import Json.Decode.Extra exposing (when)
+import Json.Decode.Extra as DecodeE exposing (when)
 import List.Extra as ListE
 import Set exposing (Set)
 import Syntax exposing (Syntax)
+import TreePath exposing (TreePath)
 import UI
 import UI.Icon as Icon
 
 
+type EmbeddedSource
+    = EmbeddedSource Syntax Syntax
+    | Builtin Syntax
```

Comment:
This should be merged into `TermSource` and `TypeSource` in a subsequent effort.